File: array_and_hashing/swift/replaceElements.swift
func replaceElements(_ arr: [Int]) -> [Int] {
    var res = Array(repeating: -1, count: arr.count)
    var maxSoFar = arr[arr.count - 1]
    for i in stride(from: arr.count - 2, through: 0, by: -1){
        res[i] = maxSoFar
        maxSoFar = max(maxSoFar, arr[i])
    }
    return res
}

File: array_and_hashing/python/replaceElements.py
def replaceElements(self, arr: List[int]) -> List[int]:
    res = [-1] * len(arr)
    maxSoFar = arr[len(arr) - 1]
    for i in range(len(arr) - 2, -1, -1):
        res[i] = maxSoFar
        maxSoFar = max(maxSoFar, arr[i])
    return res